jerry-core/vm/vm.c

```diff
@@ -3464,6 +3464,7 @@ vm_execute (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
 
   JERRY_CONTEXT (status_flags) &= (uint32_t) ~ECMA_STATUS_DIRECT_EVAL;
 
+  vm_frame_ctx_t* prev_context_p = JERRY_CONTEXT (vm_top_context_p);
```

Comment:
What do we gain exactly? Total stack consumption seems the same.

Reply:
These are the benchmark results, I think the PR has measurable effect.

Benchmark | Stack (bytes) | Perf (sec) |
----: | ----: | ----: | 
3d-cube.js | 15440 -> 15000 : +2.850% | 0.784 -> 0.786 : -0.220% | 
3d-raytrace.js | 2740 -> 2676 : +2.336% | 0.946 -> 0.952 : -0.611% | 
access-binary-trees.js | 3200 -> 3112 : +2.750% | 0.510 -> 0.510 : +0.180% | 
access-fannkuch.js | 1584 -> 1584 : 0.000% | 1.874 -> 1.869 : +0.269% | 
access-nbody.js | 1560 -> 1560 : 0.000% | 1.018 -> 1.013 : +0.562% | 
bitops-3bit-bits-in-byte.js | 1528 -> 1528 : 0.000% | 0.474 -> 0.473 : +0.098% | 
bitops-bits-in-byte.js | 1528 -> 1528 : 0.000% | 0.646 -> 0.645 : +0.049% | 
bitops-bitwise-and.js | 1304 -> 1304 : 0.000% | 0.811 -> 0.814 : -0.428% | 
bitops-nsieve-bits.js | 1528 -> 1528 : 0.000% | 1.087 -> 1.086 : +0.090% | 
controlflow-recursive.js | 67720 -> 65672 : +3.024% | 0.326 -> 0.327 : -0.334% | 
crypto-aes.js | 2368 -> 2336 : +1.351% | 0.786 -> 0.792 : -0.721% | 
crypto-md5.js | 1860 -> 1812 : +2.581% | 0.528 -> 0.526 : +0.415% | 
crypto-sha1.js | 1640 -> 1640 : 0.000% | 0.532 -> 0.528 : +0.801% | 
date-format-tofte.js | 2040 -> 2008 : +1.569% | 0.713 -> 0.710 : +0.420% | 
date-format-xparb.js | 2488 -> 2464 : +0.965% | 0.495 -> 0.495 : +0.069% | 
math-cordic.js | 1528 -> 1528 : 0.000% | 1.099 -> 1.094 : +0.466% | 
math-partial-sums.js | 1488 -> 1488 : 0.000% | 0.639 -> 0.638 : +0.040% | 
math-spectral-norm.js | 1540 -> 1528 : +0.779% | 0.467 -> 0.466 : +0.234% | 
string-base64.js | 1560 -> 1560 : 0.000% | 1.325 -> 1.326 : -0.021% | 
string-fasta.js | 1512 -> 1512 : 0.000% | 1.201 -> 1.193 : +0.689% | 
Geometric mean: | +0.917% | +0.103% | 

Binary (bytes) | master(8e39fbc46f) | patch(2f7ad7a6e7) | Diff |
----: | ----: | ----: | ----: |
size | 132664 | 132664 | 0 bytes |
.rodata | 11260 | 11251 | -9 bytes |
.dynstr | 262 | 262 | 0 bytes |
.rel.plt | 176 | 176 | 0 bytes |
.interp | 25 | 25 | 0 bytes |
.dynsym | 384 | 384 | 0 bytes |